url.go

```diff
@@ -151,18 +152,35 @@ func ParseReference(url string) (*Reference, error) {
 }
 
 func parseReference(url string) (*Reference, error) {
+	// Check if we're dealing with a v1 or v2 URL.
+	u, err := gourl.Parse(url)
+	if err != nil {
+		return nil, fmt.Errorf("entity URL is not a valid url: %s", url)
+	}
+	// Shortcut old-style URLs.
+	if u.Opaque != "" {
```

Comment:
Cool, I always saw Opaque in passing and didn't know quite what it was. Now I know! Very useful for our purposes.